### PR TITLE
Presenter factory bootstrap

### LINKED ISSUE #27: Automatic detection of application mapping
Maybe add option applicationConfig to specify config file to load mapping config from? Just parse it as .neon and load application.mapping key?

### PR DESCRIPTION
**Pull request by lulco**
Resolves https://github.com/efabrica-team/phpstan-latte/issues/27

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -205,6 +205,18 @@ parameters:
...
+### presenterFactoryBootstrap
+Type: `string`
+
+If provided return value of this php file is used as Presenter Factory.
+
+Example:
+```neon
+parameters:
+    latte:
+        engineBootstrap: presenter.factory.php
```
Comment (MartinMystikJonas) [inline docs/configuration.md]: Wrong parameter

File: src/LinkProcessor/PresenterFactoryFaker.php [hunk 2]
```diff
@@ -4,25 +4,42 @@
...
+        $this->presenterFactoryBootstrap = $presenterFactoryBootstrap;
     }
 
-    public function getPresenterFactory(): PresenterFactory
+    public function getPresenterFactory(): ?IPresenterFactory
     {
-        $presenterFactory = new PresenterFactory();
-        $presenterFactory->setMapping($this->mapping);
-        return $presenterFactory;
+        $presenterFactory = null;
```
Comment (MartinMystikJonas) [inline src/LinkProcessor/PresenterFactoryFaker.php:34]: If presenter factory is already created it should be returned amd not created again

Comment (lulco) [reply src/LinkProcessor/PresenterFactoryFaker.php:34]: Yeah, I have presenterFactory property, but only initialized, not used 🤦

File: docs/configuration.md (hunk 1 above)
Comment (lulco) [reply docs/configuration.md]: thanks

**Review by MartinMystikJonas — APPROVED** [review]

**Comment by spaze** [comment]
Can you add some more documentation please? Like when would I want to use it? And if I do, do I still need to use `applicationMapping`? Also an example of what should the file look like would help, like the example file contents in `engineBootstrap` description. Thanks!

**Comment by lulco** [comment]
Updated in https://github.com/efabrica-team/phpstan-latte/pull/307

**Pull request merged**